Remove support for client per-RoC working data

gavin-norman-sociomantic opened the pull request:
It is unused by any known request implementation and just adds complexity to the request code.

nemanja-boric-sociomantic commented:
Travis is red in D2:

```
test/neo/client/Client.d(37): Error: module swarm.neo.client.IRequestSet import 'IRequestWorkingData' not found
```

Commit:
Remove support for client per-RoC working data

It is unused by any known request implementation and just adds complexity
to the request code.

gavin-norman-sociomantic commented:
Updated.

nemanja-boric-sociomantic commented:
This change LGTM. One question: are we going to bother marking this deprecated in v4.x.x? I fear that would trigger way to many deprecations, though.

gavin-norman-sociomantic commented:
I wouldn't bother. It's not a user-facing feature, so it only affects *proto maintainers.

nemanja-boric-sociomantic commented:
Ok, let's do it.
